tests/mock-ua.c

```diff
@@ -39,11 +40,64 @@ static void update_status(JsonObject *status) {
 }
 
 static int attach(int argc, char **argv) {
-  if (argc != 1) {
-    g_printerr("Invalid args\n");
+  g_autoptr(GOptionContext) options_context = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autofree char *config_file = NULL;
+  g_autofree char *format = NULL;
+  g_autofree char *token = NULL;
+  gboolean no_auto_enable = FALSE;
+
+  const GOptionEntry options_entries[] = {
+      {"no-auto-enable", 0, 0, G_OPTION_ARG_NONE, &no_auto_enable, NULL, NULL},
+      {"attach-config", 0, 0, G_OPTION_ARG_FILENAME, &config_file, NULL, NULL},
+      {"format", 0, 0, G_OPTION_ARG_STRING, &format, NULL, NULL},
+      {NULL},
+  };
+
+  options_context = g_option_context_new("<token> [flags]");
+  g_option_context_add_main_entries(options_context, options_entries, NULL);
+
+  if (!g_option_context_parse(options_context, &argc, &argv, &error)) {
+    g_critical("Options parse error: %s", error->message);
     return EXIT_FAILURE;
   }
-  const char *token = argv[0];
+
+  g_assert_cmpint(argc, >, 1);
+  if (argc > 2)
+    token = g_strdup(argv[2]);
+
+  if (config_file) {
+    g_autofree char *config_file_contents = NULL;
+    g_autoptr(GRegex) yaml_regex = NULL;
+    g_autoptr(GMatchInfo) info = NULL;
+
+    if (!g_file_get_contents(config_file, &config_file_contents, NULL,
+                             &error)) {
+      g_critical("Cannot open config file %s: %s", config_file, error->message);
+      return EXIT_FAILURE;
+    }
+
+    // Ensure the file is only readable by the current user.
+    struct stat statbuf;
+    g_assert_cmpint(stat(config_file, &statbuf), ==, 0);
+    g_assert_cmpint(statbuf.st_mode, ==, S_IFREG | 0600);
+
+    yaml_regex = g_regex_new("^token:\\s*(\\w+)$",
+                             G_REGEX_OPTIMIZE | G_REGEX_MULTILINE, 0, NULL);
+
+    if (!g_regex_match(yaml_regex, config_file_contents, 0, &info)) {
+      g_critical("Can't find token in config file %s", config_file);
+      return EXIT_FAILURE;
+    }
+
+    for (unsigned i = 0; i < g_match_info_get_match_count(info); ++i) {
+      g_autofree char *match = g_match_info_fetch(info, i);
+      g_free(token);
+      token = g_steal_pointer(&match);
```

Comment:
Isn't this just getting the last match, couldn't this just be `g_match_info_fetch(info, g_match_info_get_match_count(info) - 1)`?

Reply:
Sure thing